chore(docs): add Clodui deployment guide (#23587)
gatsbyjs/gatsby: docs/docs/deploying-to-clodui.md

---
title: Deploying to Clodui
---

[Clodui] is a static hosting service; Clodui automatically optimizes your website files for speed. This guide explains how to deploy your Gatsby website to Clodui.

> 

Reviewer: Can we add a prerequisite section stating that this guide assumes you have a gatsby site set up and are ready to deploy?

Author: thanks for pointing that I'll add it as a prerequisite

> Below is the list of features Clodui support to improve your website speed and development efforts.

Author: @Ekwuno thanks, I'll update the PR with suggested changes

> - Your website served from a global CDN to reduce network latency.
> - Compress your files to Brotli or GZip and serve it based on the browser support.
> - Minifies all your HTML, CSS and JavaScript files.
> - Generates [WebP](https://developers.google.com/speed/webp) version of all your images. Images then served based on browser support.
> - Dynamic image resizes support based on `Client-Hint` headers or query string parameters.
> - Automatic form submission handling with support for webhook integration.
> - Deployments are atomic with support for instant rollback.

Reviewer: ```suggestion
- Serves a website from a global CDN to reduce network latency.
- Compresses files to Brotli or GZip and serve it based on the browser support.
- Minifies all your HTML, CSS, and JavaScript files.
- Generates [WebP](https://developers.google.com/speed/webp) version of all your images. Images then served based on browser support.
- Handles dynamic image resizes support based on `Client-Hint` headers or query string parameters.
- Automatic form submission handling with support for webhook integration.
- Deployments are atomic with support for instant rollback.
```

Author: @Ekwuno thanks, I'll update PR with these changes

## Steps to get started

Follow these steps to deploy your website to Clodui.

1. [Sign up](https://app.clodui.com/auth/signup) to create an account using your email and password.  

>    _**Note :** Don't use social login like Google or Facebook to sign up because Clodui CLI doesn't support it yet._

Reviewer: I am curious as to if a PR will be sent to update this not when Clodui CLI supports social login.  I would advise maybe taking the social login Option out of https://app.clodui.com/auth/signup as users might miss the note.

Author: Removing social login from the main application (https://app.clodui.com) is not an option. It is added to reduce friction in signup flow. In future, we're planning support Personal Access Token(PAT) to use in CLI. We will be sending a new PR to update the docs when PAT support is available.

2. Install the [Clodui CLI]:

>    ```bash

Reviewer: ```suggestion
   ```shell
```

   npm install @clodui/cli -g
   ```

> 3. Add environment variables `CLODUI_USERNAME`, `CLODUI_PASSWORD` and assign with Clodui username and password respectively.

Reviewer: I assume this is being added to a `.env` file in the root of a gatsby site? I would recommend stating that. Also, where is this added? Gatsby config? a code block would help.

Author: I'll include the code snippet which shows this; basically, on Linux/macOS you do like this
```
export CLODUI_USERNAME=<username>
export CLODUI_PASSWORD=<password>
```

4. To create your Clodui website from Gatsby output directory `public`; run this command from the root of your Gatsby project:

>    ```bash

Reviewer: ```suggestion
   ```shell
```

   clodui website create --source-dir public
   ```

   The command output shows website id, website name, deployment status and URL to your website. URL to your website is of the following format `https://<website-name>.clodui.com`. URL is accessible after the deployment completes, i.e. deployment status is `Deployed`.

   If the deployment is in progress, wait for the deployment to finish. To check the latest website status, run the following command.

>    ```bash

Reviewer: ```suggestion
   ```shell
```

   clodui website status --website-id <Clodui website id>
   ```

5. After your website created, to deploy new changes from `public` directory, run the following command.

>    ```bash

Reviewer: ```suggestion
   ```shell
```

---
title: Deploying to Clodui
---

[Clodui] is a static hosting service; Clodui automatically optimizes your website files for speed. This guide explains how to deploy your Gatsby website to Clodui.

Clodui supports the following features to improve your website speed and development efforts :

- Serves a website served from a global CDN to reduce network latency.
- Compresses files to Brotli or GZip and serve it based on the browser support.
- Minifies all your HTML, CSS and JavaScript files.
- Generates [WebP](https://developers.google.com/speed/webp) version of all your images. Images then served based on browser support.
- Handles dynamic image resizes based on `Client-Hint` headers or query string parameters.
- Automatic form submission handling with support for webhook integration.
- Deployments are atomic with support for instant rollback.

**Prerequisites :**

> - Your Gatsby website is setup and ready to deploy

Reviewer: ```suggestion
- This guide asummes that you have a Gatsby site setup and ready to deploy
```

> - This guide asummes that you have a Gatsby site setup and ready to deploy

Reviewer: ```suggestion
- This guide assumes that you have a Gatsby site setup and ready to deploy 
```

## Steps to get started

Follow these steps to deploy your website to Clodui.

1. [Sign up](https://app.clodui.com/auth/signup) to create an account using your email and password.  
   _**Note :** Don't use social login like Google or Facebook to sign up because Clodui CLI doesn't support it yet._
2. Install the [Clodui CLI]:
   ```shell
   npm install @clodui/cli -g
   ```
3. Set environment variables `CLODUI_USERNAME`, `CLODUI_PASSWORD` with Clodui username and password respectively.
   In Linux/macOS terminal this can be done by:

   ```shell
   export CLODUI_USERNAME=<Clodui username>
   export CLODUI_PASSWORD=<Clodui password>
   ```

   On Windows command prompt this can be done by:

>    ```batch

Reviewer: ```suggestion
   ```shell
```

   SET CLODUI_USERNAME=<Clodui username>
   SET CLODUI_PASSWORD=<Clodui password>
   ```

4. To create your Clodui website from Gatsby output directory `public`; run this command from the root of your Gatsby project:

   ```shell
   clodui website create --source-dir public
   ```

>    The command output shows website id, website name, deployment status and URL to your website. URL to your website is of the following format `https://<website-name>.clodui.com`. URL is accessible after the deployment completes, i.e. deployment status is `Deployed`.

Reviewer: ```suggestion
   The command output shows website id,  name(which is your website name), deployment status and URL to your website. URL to your website is of the following format `https://<website-name>.clodui.com`. URL is accessible after the deployment completes, i.e. deployment status is `Deployed`.
```
